nui/src/main/java/org/terasology/nui/canvas/CanvasImpl.java

```diff
@@ -171,13 +176,20 @@ public void postRender() {
 
     @Override
     public void processMousePosition(Vector2i position) {
+        processMousePosition(position, 0);
+    }
+
+    @Override
+    public void processMousePosition(Vector2i position, int pointer) {
+        Set<InteractionRegion> newMouseOverRegions = Sets.newLinkedHashSet();
```

Comment:
I think this is the same as before, but does this create a new set every time this method is call (which I guess is quite often)? 

(not changing it in this PR is good, just wanted to leave my thought here)

Reply:
This is the same as before, just moved to the top of the method. I tried to ensure that the code was functionally identical to before when using only a single pointer. Anything else was likely to cause further issues that could manifest themselves practically anywhere, since this code runs at the core of NUI. I think I tried to reduce memory allocations here in one of the earlier (unpublished) iterations of this code but it just caused far too many bugs due to accidentally-retained state.

I ran NUI under the Android Profiler to measure memory consumption and it doesn't seem too bad right now. GC collects ~4MB at intervals but memory consumption for a basic sample app ([NUISamples](https://github.com/BenjaminAmos/NUISamples)) is only around 64MB most of the time. More allocations seemed to be done when drawing UI widgets than in this method. I'll profile Destination Sol later to double check this though.

Reply:
Destination Sol doesn't seem to have any more issues with memory. It consumes around 100-120MB but NUI is a large factor in those memory allocations. It may actually be the largest factor. GC collections tend to keep that memory usage quite steady though.

Reply:
Thanks for the profiling and info on this! :green_heart: